### ISSUE TITLE
prefer-const reporting in try catch

### ISSUE THREAD

**Issue by edi9999** [eslint/eslint#8723]
**Tell us about your environment**

* **ESLint Version:** 4.0.0
* **Node Version:** 8.1.0
* **npm Version:** 5.0.3

**What parser (default, Babel-ESLint, etc.) are you using?** babel-eslint

**Please show your full configuration:**

```
{
  "parser": "babel-eslint",
  "plugins": [
    "dependencies"
  ],
  "rules": {
    "prefer-const": 2
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
let result;
try {
	result = window.bar();
}
catch (error) {
	console.log(error);
}
console.log(result);
```

**What did you expect to happen?**

No error to be thrown. It doesn't throw an error with 3.19.0

**What actually happened? Please include the actual, raw output from ESLint.**

```
/home/edgar/projects/dx/xt/es6/lexer.js
  3:2  error  'result' is never reassigned. Use 'const' instead  prefer-const
```

**Comment by not-an-aardvark** [comment]
Thanks for the report. Does this issue happen when you don't use `babel-eslint`?

**Comment by edi9999** [comment]
Indeed, it doesn't happen with : 

```
{
  "parserOptions" : {
    "ecmaVersion": 7
  },
  "plugins": [
    "dependencies"
  ],
  "rules": {
    "prefer-const": 2
  }
}
```

**Comment by not-an-aardvark** [comment]
Ok, thanks for double-checking. In that case I suspect it's a problem with [babel-eslint](https://github.com/babel/babel-eslint) -- I think it makes some monkeypatches to variable scoping, and we changed that a bit in 4.0.0.

**Comment by not-an-aardvark** [comment]
Closing because this isn't an issue with eslint, but I would recommend opening an issue on babel-eslint so they can address this.
